src/openforms/contrib/auth_oidc/plugin.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import annotations
+
+from typing import ClassVar, Protocol, TypedDict
+
+from django.http import (
+    HttpRequest,
+    HttpResponseBadRequest,
+    HttpResponseBase,
+    HttpResponseRedirect,
+)
+
+from digid_eherkenning.oidc.models.base import BaseConfig
+from mozilla_django_oidc_db.utils import do_op_logout
+from mozilla_django_oidc_db.views import _RETURN_URL_SESSION_KEY
+
+from openforms.authentication.base import BasePlugin, CosignSlice
+from openforms.authentication.constants import (
+    CO_SIGN_PARAMETER,
+    FORM_AUTH_SESSION_KEY,
+    AuthAttribute,
+)
+from openforms.authentication.exceptions import InvalidCoSignData
+from openforms.authentication.registry import register
+from openforms.authentication.typing import FormAuth
+from openforms.forms.models import Form
+from openforms.typing import JSONObject, StrOrPromise
+from openforms.utils.urls import reverse_plus
+
+from .constants import OIDC_ID_TOKEN_SESSION_KEY
+
+
+def get_config_to_plugin() -> dict[type[BaseConfig], OIDCAuthentication]:
+    """
+    Get the mapping of config class to plugin identifier from the registry.
+    """
+    return {
+        plugin.config_class: plugin
+        for plugin in register
+        if isinstance(plugin, OIDCAuthentication)
+    }
+
+
+class OptionsT(TypedDict):
+    pass
+
+
+class AuthInit(Protocol):
+    def __call__(
+        self, request: HttpRequest, return_url: str, options: OptionsT, *args, **kwargs
+    ) -> HttpResponseBase: ...
+
+
+# can't bind T to JSONObject because TypedDict and dict[str, ...] are not considered
+# assignable... :(
+class OIDCAuthentication[T, OptionsT](BasePlugin[OptionsT]):
+    verbose_name: StrOrPromise = ""
```

Comment:
I don't understand why this override is needed? The base class sets a message already.